client/src/pages/ProtectedRoute.jsx

```diff
@@ -0,0 +1,14 @@
+import { Navigate } from "react-router-dom";
+import { useAppContext } from "../context/appContext";
+
+const ProtectedRoute = ({ children }) => {
+  const { isAuthenticated } = useAppContext();
+
+  if (!isAuthenticated) {
+    return <Navigate to="/" />;
+  }
```

Comment:
You figured out how to protect the route from non-users! Looking forward to hearing more about it!